source/s3_auto_ranged_put.c

```diff
@@ -1711,6 +1711,17 @@ static int s_s3_auto_ranged_put_pause(
      */
     aws_s3_meta_request_set_fail_synced(meta_request, NULL, AWS_ERROR_S3_PAUSED);
 
+    while (!aws_linked_list_empty(&meta_request->synced_data.ongoing_http_request_list)) {
```

Comment:
nit: this can probably be pulled into a helper in meta req, so we dont have to repeat code twice